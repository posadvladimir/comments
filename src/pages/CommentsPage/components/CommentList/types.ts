
import type { TCommentsDataForRender } from '../../common/ts/types/comments';

export type TCommentListProps = {
    commentsData: TCommentsDataForRender;
}