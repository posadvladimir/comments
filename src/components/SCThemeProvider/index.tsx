
import { ThemeProvider } from 'styled-components';

import type { TChildrenProps } from 'src/ts/types/children';

import theme from 'src/styles/theme';

const SCThemeProvider: React.FC<TChildrenProps> = ({ children }) => (
    <ThemeProvider theme={theme}>
        {children}
    </ThemeProvider>
);

export default SCThemeProvider;