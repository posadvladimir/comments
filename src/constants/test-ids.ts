
const LOADER_TEST_ID = 'loader-test-id';
const BUTTON_TEST_ID = 'button-test-id';
const LIKE_COUNTER_TEST_ID = 'like-counter-test-id';
const LIKE_COUNTER_BUTTON_TEST_ID = 'like-counter-button-test-id';
const LIKE_COUNTER_TOTAL_TEST_ID = 'like-counter-total-test-id';
const COMMENT_TEST_ID = 'comment-test-id';
const ERROR_TEST_ID = 'error-test-id';

export {
    LOADER_TEST_ID,
    BUTTON_TEST_ID,
    LIKE_COUNTER_TEST_ID,
    LIKE_COUNTER_BUTTON_TEST_ID,
    COMMENT_TEST_ID,
    LIKE_COUNTER_TOTAL_TEST_ID,
    ERROR_TEST_ID,
};