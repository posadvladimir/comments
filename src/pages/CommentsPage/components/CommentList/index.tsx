
import type { TCommentListProps } from './types';

import Comment from '../Comment';

const CommentList: React.FC<TCommentListProps> = ({
    commentsData,
}) => (
    <>
        {commentsData.map(({ id, ...props }) => (
            <Comment
                key={id}
                {...props}
            />
        ))}
    </>
);

export default CommentList;