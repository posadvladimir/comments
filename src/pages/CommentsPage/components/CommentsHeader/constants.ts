
import type { TDeclensionArray } from '../../common/ts/types/declensions';

export const COMMENT_DECLENSIONS: TDeclensionArray = [
    'комментарий',
    'комментария',
    'комментариев'
];