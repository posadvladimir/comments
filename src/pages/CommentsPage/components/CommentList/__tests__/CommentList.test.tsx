

import { render, screen } from '@testing-library/react';

import { COMMENT_TEST_ID } from 'src/constants/test-ids';

import { runSnapshotTest } from 'src/common/tests/snapshot';

import SCThemeProvider from 'src/components/SCThemeProvider';
import CommentList from '..';

const testData = [
    {
        id: 1,
        authorName: 'Han Solo',
        authorAvatar: 'han-solo.jpg',
        likes: 5,
        text: 'Piu piu',
        creationTime: '2 часа назад',
        parent: null,
    },
    {
        id: 2,
        authorName: 'Yoda',
        authorAvatar: 'yoda.jpg',
        likes: 15,
        text: 'Noo',
        creationTime: '1 час назад',
        parent: 1,
    },
    {
        id: 3,
        authorName: 'Darth Wader',
        authorAvatar: 'dart-hwader.jpg',
        likes: 115,
        text: 'Ok',
        creationTime: '3 чаcа назад',
        parent: null,
    },
];

const commentListComponent = (
    <SCThemeProvider>
        <CommentList commentsData={testData} />
    </SCThemeProvider>
);

describe('Компонент Comment', () => {
    test('отображает комментарии корректно', () => {
        render(commentListComponent);

        const commentElements = screen.getAllByTestId(COMMENT_TEST_ID);

        expect(commentElements.length).toBe(testData.length);
    });

    runSnapshotTest(commentListComponent);
});