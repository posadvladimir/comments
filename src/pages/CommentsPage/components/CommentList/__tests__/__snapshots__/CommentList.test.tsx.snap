// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Comment соотвествует снепшоту 1`] = `
Array [
  <div
    className="sc-grXZZQ bVYjzh"
    data-testid="comment-test-id"
  >
    <div
      className="sc-dkjKgF gkIFfb"
    >
      <img
        alt="author avatar"
        className="sc-hLclGa gHpfka"
        src="han-solo.jpg"
      />
      <div
        className="sc-eDnVMP lgZzqD"
      >
        <span
          className="sc-bcPKhP jtwZFa"
        >
          Han Solo
        </span>
        <span
          className="sc-jSwlEQ cRyZix"
        >
          2 часа назад
        </span>
      </div>
      <div
        className="sc-iBAaJG bpbcyD"
        data-testid="like-counter-test-id"
      >
        <button
          className="sc-ftLKQv hnFVgX"
          data-testid="like-counter-button-test-id"
          onClick={[Function]}
        >
          <img
            alt="like icon"
            className="sc-pqitP fwnPig"
            src="red-stroke.png"
          />
        </button>
        <span
          className="sc-jrkPcm cQvrov"
          data-testid="like-counter-total-test-id"
        >
          5
        </span>
      </div>
    </div>
    <p
      className="sc-gKHVLF fNJWzz"
    >
      Piu piu
    </p>
  </div>,
  <div
    className="sc-grXZZQ SEcdY"
    data-testid="comment-test-id"
  >
    <div
      className="sc-dkjKgF gkIFfb"
    >
      <img
        alt="author avatar"
        className="sc-hLclGa gHpfka"
        src="yoda.jpg"
      />
      <div
        className="sc-eDnVMP lgZzqD"
      >
        <span
          className="sc-bcPKhP jtwZFa"
        >
          Yoda
        </span>
        <span
          className="sc-jSwlEQ cRyZix"
        >
          1 час назад
        </span>
      </div>
      <div
        className="sc-iBAaJG bpbcyD"
        data-testid="like-counter-test-id"
      >
        <button
          className="sc-ftLKQv hnFVgX"
          data-testid="like-counter-button-test-id"
          onClick={[Function]}
        >
          <img
            alt="like icon"
            className="sc-pqitP fwnPig"
            src="red-stroke.png"
          />
        </button>
        <span
          className="sc-jrkPcm cQvrov"
          data-testid="like-counter-total-test-id"
        >
          15
        </span>
      </div>
    </div>
    <p
      className="sc-gKHVLF fNJWzz"
    >
      Noo
    </p>
  </div>,
  <div
    className="sc-grXZZQ bVYjzh"
    data-testid="comment-test-id"
  >
    <div
      className="sc-dkjKgF gkIFfb"
    >
      <img
        alt="author avatar"
        className="sc-hLclGa gHpfka"
        src="dart-hwader.jpg"
      />
      <div
        className="sc-eDnVMP lgZzqD"
      >
        <span
          className="sc-bcPKhP jtwZFa"
        >
          Darth Wader
        </span>
        <span
          className="sc-jSwlEQ cRyZix"
        >
          3 чаcа назад
        </span>
      </div>
      <div
        className="sc-iBAaJG bpbcyD"
        data-testid="like-counter-test-id"
      >
        <button
          className="sc-ftLKQv hnFVgX"
          data-testid="like-counter-button-test-id"
          onClick={[Function]}
        >
          <img
            alt="like icon"
            className="sc-pqitP fwnPig"
            src="red-stroke.png"
          />
        </button>
        <span
          className="sc-jrkPcm cQvrov"
          data-testid="like-counter-total-test-id"
        >
          115
        </span>
      </div>
    </div>
    <p
      className="sc-gKHVLF fNJWzz"
    >
      Ok
    </p>
  </div>,
]
`;
